app/forms/admin/adjustments/create_adjustment_form.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Admin::Adjustments
+  class CreateAdjustmentForm
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :session
+    attribute :statement
+    attribute :description
+    attribute :amount, :integer
+    attribute :adjustment
+
+    validate :adjustment_valid
+    validate :statement_open
+
+    def initialize(*)
+      super
+      session[:created_adjustment_ids] ||= []
```

Comment:
This shouldn't need know about the session - its a model not a controller and this breaks separation of concerns - surely it can just get initialized with the adjustment_ids themselves and the controller be responsible for passing in the ids (which it can choose to hold in the session)

and have `#save` update the attribute with list of adjustment ids? Controller again can update whereever it holds it, eg session

Reply:
Thinking some more on this overnight, ultimately the model is just being handed a hash (from its perspective) to persist into. It just happens to be a session, so a more minimal change would be to rename this to something like `store` instead of `session` 

I still think its better to have the controller putting stuff into the session, instead of the model but other models do back onto other stores (eg AR -> postgres)

Reply:
I've gone for the first option - just maintaining an array of created_adjustment_ids in the form.
The controller is now responsible for storing it in the session, as well as clearing it.